lighthouse-cli/test/smokehouse/report-assert.js

```diff
@@ -175,7 +175,7 @@ function collateResults(localConsole, actual, expected) {
     const artifactNames = /** @type {(keyof LH.Artifacts)[]} */ (Object.keys(expectedArtifacts));
     artifactAssertions = artifactNames.map(artifactName => {
       const actualResult = (actual.artifacts || {})[artifactName];
-      if (!actualResult) {
+      if (!actualResult && expectedArtifacts[artifactName]) {
```

Comment:
what case is this handling? This doesn't seem possible except in avoidable situations

Reply:
![image](https://user-images.githubusercontent.com/4071474/85806126-0c74fd80-b703-11ea-929d-5f438df58923.png)

Reply:
`Object.keys(expectedArtifacts)` means an expectation is defined (even if the value is falsy), so maybe this conditional should switch to checking the same but for `artifactName` in `actual.artifacts`?

i.e. if the expectations have a `FullPageScreenshot: null` we should still warn if the actual has nothing at all defined.

Reply:
Oh, it still warns. this is just suppressing printing out `Config run did not generate artifact...` when the expectation is that it emits a falsy artifact.

maybe it should be `!(artifactName in actualArtifacts)`?

Reply:
> maybe it should be `!(artifactName in actualArtifacts)`?

👍  yes, that's what I mean